engine/src/main/java/org/hibernate/search/annotations/Facet.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Extension annotation for {@code @Field} enabling it for faceting.
+ *
+ * @author Hardy Ferentschik
+ * @hsearch.experimental : This feature is experimental
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD, ElementType.FIELD })
+@Documented
+public @interface Facet {
+
+	/**
+	 * @return the facet name. Defaults to the name of the field this facet refers to
+	 */
+	String name() default "";
+
+	/**
+	 * @return the field name this annotation refers to. It can be omitted in case there is only a single {@code @Field}
+	 * annotation
+	 */
+	String forField() default "";
+
+	/**
+	 * @return the encoding type to use for this facet. Per default the encoding type is chosen based on the type of the
+	 * entity property.
+	 */
+	FacetEncodingType encoding() default FacetEncodingType.AUTO;
```

Comment:
Discrete vs Range facets. Per default strings are discrete (distinct values) whereas numbers are numeric doc values which can be used for ranges. Now think about a case where your domain contains a int value, but the values are really just a set of distinct values (or at least you want to treat them as such).  For example, a rating 1-10. You can also work with ranges here of course, but if you just want the count for each of the different rating a discrete facet (using string encoding) is much simpler). Does this make sense? The documentation will contain some details about this of course.